### PR TITLE
Update ExecutorClassLoader.scala

### PR DESCRIPTION
**Pull request by guolianwei**
Error when starting spark shell in windows: 
Caused by: java.net.URISyntaxException: Illegal character in path at index 32: spark://LAPTOP-XXX:XXX/X:\classes
        at java.net.URI$Parser.fail(URI.java:2848)
        at java.net.URI$Parser.checkChars(URI.java:3021)
        at java.net.URI$Parser.parseHierarchical(URI.java:3105)
        at java.net.URI$Parser.parse(URI.java:3053)
        at java.net.URI.<init>(URI.java:588)
        at org.apache.spark.repl.ExecutorClassLoader.<init>(ExecutorClassLoader.scala:57)

### CODE REVIEW

**Comment by wangyum** [comment]
@guolianwei Thank you for your contribution. Could you enable GitHub Actions to run test.

File: repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala [hunk 1]
```diff
@@ -54,7 +54,7 @@ class ExecutorClassLoader(
     classUri: String,
     parent: ClassLoader,
     userClassPathFirst: Boolean) extends ClassLoader(null) with Logging {
-  val uri = new URI(classUri)
+  val uri = new URI(classUri.replace('\\', '/'))
```
Comment (dongjoon-hyun) [inline repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala:57]: Hi, @guolianwei . We use `Utils.isWindows` to use Window-specific code.

https://github.com/apache/spark/blob/b6aea1a8d99b3d99e91f7f195b23169d3d61b6a7/launcher/src/main/java/org/apache/spark/launcher/Main.java#L90-L91

**Comment by dongjoon-hyun** [comment]
On top of @wangyum 's comment, please file an Apache Spark JIRA issue . You can see more contributor's guide here.
- https://spark.apache.org/contributing.html

File: repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ class ExecutorClassLoader(
     classUri: String,
     parent: ClassLoader,
     userClassPathFirst: Boolean) extends ClassLoader(null) with Logging {
-  val uri = new URI(classUri)
```
Comment (HyukjinKwon) [inline repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala:57]: @guolianwei plase

1. file a JIRA
2. describe how you reproduced the issue and tested this patch
3. add a unit test.

See also https://spark.apache.org/contributing.html

**Pull request closed without merging**